client/src/components/relatedProducts/relatedProductsList.jsx

```diff
@@ -24,43 +29,88 @@ const responsive = {
+        }))
+          .then(response => {
+            let productsInfoArray = response;
+
+            //ONCE WE HAVE ALL RELATED PRODUCTS FIND THE STYLES FOR EACH PRODUCT
+            Promise.all(productsInfoArray.map(product => {
+              return axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hr-rfe/products/${product.id}/styles`, {
+                headers: { Authorization: API_KEY }
+              })
+                .then(response => {
+                  let productStylesObj = response.data;
+
+                  return productStylesObj;
+                })
+            }))
+              .then(response => {
 
+                let productsStylesArray = response;
 
+                Promise.all(productsStylesArray.map(product => {
+                  return axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hr-rfe/reviews?product_id=${product.product_id}`, {
+                    headers: { Authorization: API_KEY }
+                  })
+                    .then(response => {
+                      let productReviewsObj = response.data
 
+                      return productReviewsObj
+                    })
+                }))
+                  .then(response => {
+                    let productReviewsArray = response
+
+                    productsInfoArray.map((product, index) => {
+                      product.styles = productsStylesArray[index];
+                      product.reviews = productReviewsArray[index];
+                    })
+
+                    setAccumulatedProductData(productsInfoArray);
+                  })
+
+              })
+          })
+      })
+  }, []);
 
 
   return (
     <Carousel responsive={responsive}>
-        <div>
-            <img src="https://image.shutterstock.com/image-vector/colorful-illustration-test-word-260nw-1438324490.jpg" />
-            <p className="legend">Legend 1</p>
-        </div>
-        <div>
-            <img src="https://image.shutterstock.com/image-vector/colorful-illustration-test-word-260nw-1438324490.jpg" />
-            <p className="legend">Legend 2</p>
-        </div>
-        <div>
-            <img src="https://image.shutterstock.com/image-vector/colorful-illustration-test-word-260nw-1438324490.jpg" />
-            <p className="legend">Legend 3</p>
-        </div>
+      {accumulatedProductData.map((product, index) =>{
+
+        let style = product.styles.results.length > 1 ?
+          product.styles.results.find(result => result['default?'] === true) !== undefined ?
+          product.styles.results.find(result => result['default?'] === true)
+          : product.styles.results[0]
+          : product.styles.results[0];
+
+          // console.log('style', style)
+          // console.log('styleUrl', style.photos[0].url)
+
+
+        return (<Card picUrl= {style.photos[0]['url']} category={product.category} name={product.name} price={style.original_price} salePrice={style.sale_price} key={index} />)
+      } )}
     </Carousel>
-);
+  );
 }
 
```

Comment:
consider separating this into multiple files for the sake of organization.

Reply:
will do.